Dockerfile

```diff
@@ -1,14 +1,46 @@
-FROM alpine:3.17 AS verify
+# Stage 1: Fetch an Alpine container so that we may download the appropriate
+#          version of Amazon Linux
+FROM alpine:3.19 AS extract
+
+# The variables set here will determine what is downloaded.
+ARG target_arch="arm64"
+
+ARG amzn_linux_release="al2023"
+ARG amzn_linux_version="2023.3.20231218.0"
+ARG amzn_linux_file="${amzn_linux_release}-container-${amzn_linux_version}-${target_arch}.tar.xz"
+ARG amzn_linux_url="https://amazon-linux-docker-sources.s3.amazonaws.com/${amzn_linux_release}/${amzn_linux_version}/${amzn_linux_file}"
+
+
+# Download the packages required to extract the files from the container source
+# archive.
 RUN apk add --no-cache curl tar xz
 
-RUN ROOTFS=$(curl -sfOJL -w "al2023-container-2023.3.20240108.0-arm64.tar.xz" "https://amazon-linux-docker-sources.s3.amazonaws.com/al2023/2023.3.20240108.0/al2023-container-2023.3.20240108.0-arm64.tar.xz") \
-  && echo 'cc56e4426e4e51cd15c943f2f3ab51adf6918b53bce5c3b4b7cdfe91b038fa28  al2023-container-2023.3.20240108.0-arm64.tar.xz' >> /tmp/al2023-container-2023.3.20240108.0-arm64.tar.xz.sha256 \
-  && cat /tmp/al2023-container-2023.3.20240108.0-arm64.tar.xz.sha256 \
-  && sha256sum -c /tmp/al2023-container-2023.3.20240108.0-arm64.tar.xz.sha256 \
-  && mkdir /rootfs \
-  && tar -C /rootfs --extract --file "${ROOTFS}"
+# The below does the following:
+#   * Downloads the compressed container archive, and the appropriate sha256sum
+#     file from the docker sources repository.
+#   * Verifies the container archive SHA256 checksum.
+#   * Extracts the content of the archive into /rootfs.
+RUN                                                   \
+  curl                                                \
+    --fail                                            \
+    --location                                        \
+    --remote-name                                     \
+    --url "${amzn_linux_url}"                         \
+&& curl                                               \
+     --fail                                           \
+     --location                                       \
+     --remote-name                                    \
+     --url "${amzn_linux_url}.sha256"                 \
+&& cat ${amzn_linux_file}.sha256                      \
```

Comment:
The checksum is included in the Dockerfile itself as a form of embedded trust store - it ensures that, if at some point in the future the contents of the external image are changed, you'll know about it because the checksum will fail.

Reply:
Yeah - I've been having a few realizations… after I submitted the PR of course ;)